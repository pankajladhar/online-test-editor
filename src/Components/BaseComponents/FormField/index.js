import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class FormField extends PureComponent {
    render() {
        return (
            <div className={`FormField ${this.props.className}`}>
                {this.props.children}
            </div>
        );
    }
}

FormField.defaultProps =  {
    className: ""
}

FormField.propTypes = {
    className: PropTypes.string,
    children: PropTypes.any.isRequired,
};

export default FormField;