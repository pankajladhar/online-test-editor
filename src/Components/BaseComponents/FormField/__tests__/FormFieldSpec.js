import React, { Children } from 'react';
import renderer from 'react-test-renderer';
import FormField from './../index';

describe('FormField', () => {
    let formFieldProps = {}
    beforeEach(() => {
        formFieldProps = {
            children: <h1>Some Children</h1>
        }
    });

    it('renders correctly with mandatory params', () => {
        let tree = renderer.create(
            <FormField {...formFieldProps} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with mandatory and optional params', () => {
        formFieldProps.className = "SomeClass";
        let tree = renderer.create(
            <FormField {...formFieldProps} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

});