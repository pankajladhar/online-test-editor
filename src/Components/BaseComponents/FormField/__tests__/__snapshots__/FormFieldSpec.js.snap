// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField renders correctly with mandatory and optional params 1`] = `
<div
  className="FormField SomeClass"
>
  <h1>
    Some Children
  </h1>
</div>
`;

exports[`FormField renders correctly with mandatory params 1`] = `
<div
  className="FormField "
>
  <h1>
    Some Children
  </h1>
</div>
`;
